### ISSUE TITLE
Best practices for code to be schema migration friendly?

### ISSUE THREAD

**Issue by OkayX6** [fsprojects/FSharp.Data.SqlClient#172]
Hi!

Let's say I have a query like this:
```SQL
SELECT * FROM [dbo].[User] WHERE UserId = @userId
```

Now let's say I am adding a new column with default value. Structurally, the code could be resistant to that change, right?
I was expecting my code to be robust to such changes but no, it crashes:

```
[ArgumentException: The arrays have different lengths.
Parameter name: array2]
   Microsoft.FSharp.Collections.ArrayModule.Zip(T1[] array1, T2[] array2) +2240920
   <StartupCode$Lost10Found>.getUser@33-4.Invoke(Object[] values) in C:\Users\Denis\Documents\Projects\Lost10Found\Lost10Found\Controllers\BaseController.fs:33
   FSharp.Data.ofReader@35.GenerateNext(IEnumerable`1& next) in C:\Users\mitekm\Documents\GitHub\FSharp.Data.SqlClient\src\SqlClient\ISqlCommand.fs:39
```

That means I have to write my queries with explicit column names like this:
```SQL
SELECT [UserId], [Name] FROM [dbo].[User] WHERE UserId = @userId
```

What is your position on that?

**Comment by dmitry-a-morozov** [comment]
I think you're right. 
Unfortunately it requires changes at very internals of implementation.
This is not an easy fix but I will try to get it done.

**Comment by dmitry-a-morozov** [comment]
fixed in v1.8
